Common/src/config_structure.cpp

```diff
@@ -33,10 +33,13 @@
 
 #include "../include/config_structure.hpp"
 
+#include "../include/ad_structure.hpp"
+
+
 CConfig::CConfig(char case_filename[MAX_STRING_SIZE], unsigned short val_software, unsigned short val_iZone, unsigned short val_nZone, unsigned short val_nDim, unsigned short verb_level) {
   
 #ifdef HAVE_MPI
-  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+  SU2_MPI::Comm_rank(MPI_COMM_WORLD, &rank);
```

Comment:
I already added a default wrapper implementation in this branch which simply copies the data for most of the mpi calls. So in theory we can already remove all the #ifdefs. However, like you said, I am not quite sure if that works for all parts, especially in geometry_structure.cpp with all the non-blocking communication. That's the reason why I haven't done it yet.

About the second aspect you are mentioning: I also thought about that. Another possibility would be to put this stuff in the wrapper. So that you can simply do SU2_MPI::Get_rank()/SU2_MPI::Get_size() or something.